### PR TITLE
Improve benchmark running script

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

Running a benchmark on a particular LRS that runs locally (or not) should be more flexible.

## Proposal

Decouple LRS configuration from the benchmarking script to make it more generic.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,8 +11,48 @@ that best fit with their needs and constraints.
...
+
+```
+usage: bin/bench LRS_ROOT_URL PROTOCOL [DURATION] [CONCURRENT_USERS] [STATEMENTS_PER_REQUEST]
+```
+
+You will find your results in the latest runs directory:
+
+```
+$ ls runs | tail -n 1
+```
```
Comment (quitterie-lcs) [inline README.md:41]: Maybe we can wrap it in a make command `make results` ?

File: bin/bench [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+declare | egrep "^LBT" > "${LBT_RUNS_DIR}/${LBT_WORKING_DIR}/parameters.txt"
+
+# Run baby run
+DOCKER_USER=${DOCKER_USER} ${COMPOSE} \
+  -f docker-compose.yml \
+  run --rm locust \
+      -f "${LBT_PROTOCOL}" \
+      -H "${LBT_LRS_ROOT_URL}" \
+      --statements-per-req "${LBT_STATEMENTS_PER_REQUEST}" \
+      --csv="/app/${LBT_RUNS_DIR}/${LBT_WORKING_DIR}/run" \
```
Comment (quitterie-lcs) [inline bin/bench:56]: When running a benchmark, this csv file is missing

File: README.md [hunk 3]
```diff
@@ -11,8 +11,48 @@ that best fit with their needs and constraints.
...
+
+You should see a bunch of CSV files created by `locust` and a `parameters.txt`
+files with used input parameters for this file:
```
Comment (wilbrdt) [inline README.md]:
```suggestion
file containing the input parameters used for this run:
```

**Review by wilbrdt — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (jmaupetit) [reply README.md:41]: :thinking: I like the idea of having a command / utility to list runs as a CSV rows:

```
$ bin/runs

users; duration; lrs_root_url; protocol; statements_per_request; run
1; 30; http://AAA:BBB@ralph:8090/xAPI/statements ; /app/locust/post.py; 1; 2023-03-31T17:00:25,608936440-04:00
1; 30; http://AAA:BBB@lrsql:8090/xAPI/statements ; /app/locust/post.py; 1; 2023-03-31T18:00:25,608926440-04:00
1; 30; http://AAA:BBB@learninglocker:8090/xAPI/statements ; /app/locust/post.py; 1; 2023-03-31T18:12:25,508936440-04:00
```

This allows cool rendering such as:

```
$ bin/runs | csvlook -I

| users |  duration |  lrs_root_url                                        |  protocol            |  statements_per_request |  run                                 |
| ----- | --------- | ---------------------------------------------------- | -------------------- | ----------------------- | ------------------------------------ |
| 1     |  30       |  http://AAA:BBB@ralph:8090/xAPI/statements           |  /app/locust/post.py |  1                      |  2023-03-31T17:00:25,608936440-04:00 |
| 1     |  30       |  http://AAA:BBB@lrsql:8090/xAPI/statements           |  /app/locust/post.py |  1                      |  2023-03-31T18:00:25,608926440-04:00 |
| 1     |  30       |  http://AAA:BBB@learninglocker:8090/xAPI/statements  |  /app/locust/post.py |  1                      |  2023-03-31T18:12:25,508936440-04:00 |
```

And we can also imagine filters such as:

```
$ bin/runs "ralph" | csvlook -I

| users |  duration |  lrs_root_url                                        |  protocol            |  statements_per_request |  run                                 |
| ----- | --------- | ---------------------------------------------------- | -------------------- | ----------------------- | ------------------------------------ |
| 1     |  30       |  http://AAA:BBB@ralph:8090/xAPI/statements           |  /app/locust/post.py |  1                      |  2023-03-31T17:00:25,608936440-04:00 |
```

Do you mind if we implement this in another PR?

File: bin/bench (hunk 2 above)
Comment (jmaupetit) [reply bin/bench:56]: Yes, it's a prefix used by Locust for the files where the results are stored, _e.g._:

```
runs/2023-03-31T17:04:04,102856292-04:00
├── parameters.txt
├── run_exceptions.csv
├── run_failures.csv
├── run_stats.csv
└── run_stats_history.csv

0 directories, 5 files
```

**Pull request merged**